hashketball.rb

```diff
@@ -1,11 +1,214 @@
+         assists: 12,
+         steals: 4,
+         blocks: 5,
+         slam_dunks: 5},
+         
+         {player_name: "Ben Gordon",
+         number: 8,
+         shoe: 15,
+         points: 33,
+         rebounds: 3,
+         assists: 2,
+         steals: 1,
+         blocks: 1,
+         slam_dunks: 0},
+         
+         {player_name: "Kemba Walker",
+         number: 33,
+         shoe: 15,
+         points: 6,
+         rebounds: 12,
+         assists: 12,
+         steals: 7,
+         blocks: 5,
+         slam_dunks: 12}]
+        }
+    }
+end
 
+def num_points_scored(player)
+  game_hash.each do |(side, info_hash)|
+    info_hash[:players].each do |player_hash|
+      if player_hash[:player_name] == player
+        return player_hash[:points]
+      end
+    end
+  end
+end
 
+def shoe_size(player)
+  game_hash.each do |(side, info_hash)|
+    info_hash[:players].each do |player_hash|
+      if player_hash[:player_name] == player
+        return player_hash[:shoe]
+      end
+    end
+  end
+end
 
+def team_colors(team)
+  game_hash.each do |(side, info_hash)|
+    if info_hash[:team_name] == team
+      return info_hash[:colors]
+    end
+  end
+end
 
+def team_names
+  game_hash.map {|(side, info_hash)| info_hash[:team_name]}
+end
 
+def player_numbers(team)
+  game_hash.each do |(side, info_hash)|
+    if info_hash[:team_name] == team
+      return info_hash[:players].map {|player_hash| player_hash[:number]}
+    end
+  end
+end
+      
+def player_stats(player)
+  game_hash.each do |(side, info_hash)|
+    info_hash[:players].each do |player_hash|
+      if player_hash[:player_name] == player
+        player_hash.shift
+        return player_hash
+      end
+    end
+  end
+end
 
+def big_shoe_rebounds #find player with largest shoe size and return their rebounds
+  find_max(:shoe, :rebounds)
+end
 
+#bonus
 
+def most_points_scored
+  find_max(:points, :player_name)
+end
 
+def find_max(to_compare, to_return)
+  maximum = game_hash.reduce({}) do |memo, (side, info_hash)|
+    memo = info_hash[:players][0] if !memo[to_compare]
+    max_hash = info_hash[:players].max_by {|player_hash| player_hash[to_compare]}
+    memo = max_hash if max_hash[to_compare] > memo[to_compare]
+    memo
+  end
+  maximum[to_return]
+end
 
+def point_total(side)
+  game_hash[side][:players].reduce(0) do |total, player_hash|
+    total += player_hash[:points]
+  end
+end
+
+def winning_team
+  home_points = point_total(:home)
+  away_points = point_total(:away)
+  if home_points > away_points
+    return game_hash[:home][:team_name]
+  else
+    return game_hash[:away][:team_name]
+  end
+end
+
+def player_with_longest_name
+  find_longest = game_hash.reduce({}) do |memo, (side, info_hash)|
+    memo = info_hash[:players][0] if !memo[:player_name]
+    longest_hash = info_hash[:players].max_by {|player_hash| player_hash[:player_name].length}
+    memo = longest_hash if longest_hash[:player_name].length > memo[:player_name].length
```

Comment:
Same suggestions for cleaning this up as `find_max` above.

In theory, you could also generalize `find_max` a bit by having it `yield` the value inside the `max_by` block - then you could have the block call `value.length` here. Whether or not that would actually be worth it would depend on how often you're doing this type of operation.